client/my-sites/plan-features-2023-grid/free-plan-paid-domain-dialog.tsx

```diff
@@ -0,0 +1,230 @@
+		flex-direction: row;
+		align-items: center;
+	}
+`;
+
+const RowWithBorder = styled( Row )`
+	border-bottom: 1px solid rgba( 220, 220, 222, 0.64 );
+	padding-bottom: 16px;
+`;
+
+const DomainName = styled.div`
+	font-size: 16px;
+	line-height: 20px;
+	letter-spacing: -0.24px;
+	color: var( --studio-gray-80 );
+	overflow-wrap: break-word;
+	max-width: 100%;
+	@media ( min-width: 780px ) {
+		max-width: 55%;
+	}
+`;
+
+const FreeDomainText = styled.div`
+	font-size: 14px;
+	line-height: 20px;
+	letter-spacing: -0.15px;
+	color: var( --studio-green-50 );
+	margin-top: 4px;
+`;
+
+const StyledButton = styled( Button )`
+	padding: 10px 24px;
+	border-radius: 4px;
+	font-weight: 500;
+	font-size: 14px;
+	line-height: 20px;
+	flex: 1;
+	&.is-primary,
+	&.is-primary.is-busy,
+	&.is-primary:hover,
+	&.is-primary:focus {
+		background-color: var( --studio-blue-50 );
+		border: unset;
+	}
+	&:hover {
+		opacity: 0.85;
+		transition: 0.7s;
+	}
+	&:focus {
+		box-shadow: 0 0 0 2px var( --studio-white ), 0 0 0 4px var( --studio-blue-50 );
+	}
+	width: 100%;
+	@media ( min-width: 780px ) {
+		max-width: fit-content;
+		width: unset;
+	}
+`;
+
+const LoadingPlaceHolder = styled.div`
+	margin: 0;
+`;
+
+export function FreePlanPaidDomainDialog( {
+	domainName,
+	planSlug,
+	onFreePlanSelected,
+	onPlanSelected,
+	onClose,
+}: {
+	domainName: string;
+	planSlug: PlanSlug;
+	onClose: () => void;
+	onFreePlanSelected: ( domainSuggestion: DomainSuggestion ) => void;
+	onPlanSelected: () => void;
+} ) {
+	const translate = useTranslate();
+	const queryClient = useQueryClient();
+	const [ isBusy, setIsBusy ] = useState( false );
+	const planPrices = usePlanPrices( { planSlug, returnMonthly: true } );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const {
+		data: wordPressSubdomainSuggestion,
+		isLoading,
+		isError,
+	} = useGetWordPressSubdomain( domainName );
+	const planTitle = getPlan( planSlug )?.getTitle();
+
+	function handlePaidPlanClick() {
+		setIsBusy( true );
+		onPlanSelected();
+	}
+
+	function handleFreeDomainClick() {
+		setIsBusy( true );
+		// Since this domain will not be available after it is selected, invalidate the cache.
+		queryClient.invalidateQueries( USE_GET_WORDPRESS_SUBDOMAIN_QUERY_KEY );
+		if ( wordPressSubdomainSuggestion ) {
+			onFreePlanSelected( wordPressSubdomainSuggestion );
+		}
+	}
+
+	return (
+		<Dialog
+			isBackdropVisible={ true }
+			isVisible={ true }
+			onClose={ onClose }
+			showCloseIcon={ true }
+		>
+			<Global
+				styles={ css`
+					.dialog__backdrop.is-full-screen {
+						background-color: rgba( 0, 0, 0, 0.6 );
+					}
+					.ReactModal__Content--after-open.dialog.card {
+						border-radius: 4px;
+						width: 639px;
+					}
+				` }
+			/>
+			<DialogContainer>
+				<Heading>{ translate( 'A paid plan is required for your domain.' ) }</Heading>
+				<SubHeading>
+					{ translate(
+						'Custom domains are only available with a paid plan. And they are free for the first year.'
```

Comment:
To be very precise, we should emphasize that they are free for the first year with annual paid plans. e.g. `Custom domains are only available with a paid plan. And they are free for the first year with an annual paid plan.` It's super mouthful but I don't have a better idea for now. Also, there is a related issue that I'm outlining below :)

Reply:
How about this?
`Custom domains are only available with an annual paid plan. And they are free for the first year.`

Reply:
An annual paid plan is not required for using the custom domains as primary; just that they are only free for the first year with an annual paid plan. We still need to massage the copy here a bit :)

Reply:
You're right! I'm changing it to `Custom domains are only available with a paid plan. And they are free for the first year with an annual paid plan.` as per your suggestion, so that the copy is accurate. We can iterate on the wordiness later. 🙂